client/src/app/components/environments/environment-list/environment-list-view.component.ts

```diff
@@ -230,91 +213,45 @@ export class EnvironmentListViewComponent implements OnDestroy {
     });
   });
 
-  getFullUrl(url: string): string {
-    if (url && !url.startsWith('http') && !url.startsWith('https')) {
-      return 'http://' + url;
-    }
-    return url;
-  }
-
-  openExternalLink(event: MouseEvent, environment: EnvironmentDto): void {
-    // Prevent the click event from propagating
-    event.stopPropagation();
-
-    // Only proceed if the server URL is available
-    if (environment.serverUrl) {
-      window.open(this.getFullUrl(environment.serverUrl), '_blank');
-    }
-  }
+  private readonly environmentTypeOrder = ['test', 'staging'];
 
-  getDeploymentTime(environment: EnvironmentDto) {
-    const date = environment.latestDeployment?.updatedAt;
-    return date ? this.datePipe.transform(date, 'd MMMM y, h:mm a') : null; // Format date
-  }
+  environmentGroups = computed(() => {
```

Comment:
question: can we do that server side or is there a reason to do the grouping on the client?